games-puzzle/color-lines/patches/color_lines-0.6.patchset

```diff
@@ -0,0 +1,97 @@
+From b2d2243f354684fb8370db2f8f961b1a09cb451a Mon Sep 17 00:00:00 2001
+From: Rafal Bernacki <rav.bernacki@gmail.com>
+Date: Mon, 9 Dec 2019 12:18:37 +0000
+Subject: applying patch color_lines-0.6.patch
+
+
+diff --git a/main.c b/main.c
+index 392d09e..77dd941 100644
+--- a/main.c
++++ b/main.c
+@@ -7,6 +7,11 @@
+ #include <SDL_thread.h>
+ // #include <pthread.h>
+ 
++#ifdef __HAIKU__
++#include <unistd.h>
++#include <libgen.h>
++#endif
++
+ #ifndef WINDOWS
+ #include <unistd.h>
+ #include <pwd.h>
+@@ -1559,6 +1564,7 @@ int WINAPI WinMain (HINSTANCE hInstance,
+ #ifdef MAEMO
+ 	putenv("SDL_VIDEO_X11_WMCLASS=lines");
+ #endif
++	chdir(dirname(argv[0]));
+ 	game_mutex = SDL_CreateMutex();
+ 	if (gfx_init())
+ 		return 1;
+-- 
+2.23.0
+
+
+From eda8e5e42f6e104675fa6807f993690e32d644e5 Mon Sep 17 00:00:00 2001
+From: Ra-V2 <rav.bernacki@gmail.com>
+Date: Mon, 9 Dec 2019 12:26:40 +0000
+Subject: add new saves localization
+
+
+diff --git a/main.c b/main.c
+index 77dd941..4e79efe 100644
+--- a/main.c
++++ b/main.c
+@@ -10,6 +10,8 @@
+ #ifdef __HAIKU__
+ #include <unistd.h>
+ #include <libgen.h>
++#include <sys/types.h>
++#include <sys/stat.h>
+ #endif
+ 
+ #ifndef WINDOWS
+@@ -1636,12 +1638,17 @@ char *get_path(int n)
+ 	static char path[255];
+ 
+ #ifndef WINDOWS
++#ifdef __HAIKU__
++mkdir("/boot/home/color_lines");
```

Comment:
use find_directory to locate the B_USER_SETTINGS_DIRECTORY, and store the data there.